include/gpio.h

```diff
@@ -202,6 +202,12 @@ static inline int gpio_pin_configure(struct device *port, u32_t pin,
 
 /**
  * @brief Write the data value to a single pin.
+ *
+ * Write logical data value to a pin, taking into account GPIO_ACTIVE_LOW flag.
+ * If an output was configured as Active Low, writing value 1 to a pin will set
+ * it to low output state. Otherwise writing value 1 to a pin will set it to
+ * high output state.
```

Comment:
I am against changing the definition of how `gpio_pin_write` and `gpio_pin_read` are expected to behave. This would require alteration of all existing implementations of GPIO drivers (as the `GPIO_ACTIVE_LOW` flag cannot be just not supported).
In my opinion the `GPIO_ACTIVE_*` flags should be handled outside of particular GPIO drivers, for example in wrappers as I proposed in https://github.com/zephyrproject-rtos/zephyr/pull/11880#issuecomment-461011334.

Reply:
I provided a bit more elaborate answer in my [comment](https://github.com/zephyrproject-rtos/zephyr/pull/11880#issuecomment-461080940) in #11880. In short `GPIO_ACTIVE_LOW` is just a different name for `GPIO_POL_INV` flag which controls the value returned by `gpio_pin_write` and `gpio_pin_read` functions. Unfortunately, the documentation of these functions did not state it explicitly. The only reason we rename the `GPIO_POL_INV` flag to `GPIO_ACTIVE_LOW` is to be compatible with the definition of DTS gpio node in Linux.